test/common.py

```diff
@@ -1505,7 +1522,11 @@ def cleanup(line):
   def run_js(self, filename, engine=None, args=None,
              assert_returncode=0,
              interleaved_output=True,
-             input=None):
+             input=None,
+             run_in_tmpdir=False):
```

Comment:
I wonder if we can avoid adding this extra logic to `run_js` which is use in a lot of different places.

Could we instead allow just add `cwd=None` here and allow higher level functions to pass CWD?

Is there some fundamental reason why the tests need to run in a subdirectory?

Reply:
> Good idea! I'll try to implement this.

Done!